### DISCUSSION TITLE
Any plans to have a feature to seed values to the vault?

### DISCUSSION THREAD

**Discussion by eugbyte** [nagyesta/lowkey-vault#233] [Ideas]
Featue request to expose an endpoint, or a CLI argument, where we can seed the vault with values from a json file?

The JSON file can mimic the schema of the vault data

**Comment by nagyesta** [comment]
Absolutely! In fact, it was added in #207 
Regarding the format, this file can be a good example, but there are more in the same folder:
https://github.com/nagyesta/lowkey-vault/blob/b93c744565528e0ce0fc7ca3f637f3101218679f/lowkey-vault-app/src/test/resources/template/full-import.json.hbs

Also, if you populate your vault with some secrets and you call the relevant management endpoint (```GET /management/vault/export``` I think) then you can export the full vault using the same JSON format.

Swagger can help you with the export request: https://localhost:8443/api/swagger-ui/index.html
And this example has a Docker compose that can help with the import: https://github.com/nagyesta/spring-boot-local-key-vault-demo/blob/main/docker-compose.yml

**Reply by nagyesta** [reply]
This secion in the docs can help too: https://github.com/nagyesta/lowkey-vault/blob/main/lowkey-vault-app/README.md#importing-vault-content-at-startup

**Reply by eugbyte** [reply]
Is the file location argument `LOWKEY_IMPORT_LOCATION` relative to where the docker run command is executed?

**Reply by eugbyte** [reply]
Thanks for the help!

**Reply by eugbyte** [reply]
Also is it necessary for the json file to end with `.hbs`?

**Reply by nagyesta** [reply]
It is relative to the container root (this path is inside the container, the folder the file is in is mounted as a volume).
This ```$PWD/local/lowkey-vault/import/``` path is the path relative to the Docker compose and it is attached as ```/import``` inside the container. (It is from lines 10 and 19 in the example).

**Reply by eugbyte** [reply]
my current docker run command:
`docker run --rm -d -p 8443:8443 --name lowkey_vault nagyesta/lowkey-vault:1.8.14 --env LOWKEY_IMPORT_LOCATION=/services/notify/vault.secrets.json.hbs`

`/services/notify/vault.secrets.json.hbs` is based on the relative file path from my current working directory

**Reply by nagyesta** [reply]
> Also is it necessary for the json file to end with `.hbs`?

No, it can be json (or any extension) it is a Handlebars template, so I have used `.hbs` to indicate it, but you don't have to use the same naming and you don't need to use the template variables either.

**Reply by nagyesta** [reply]
> my current docker run command: `docker run --rm -d -p 8443:8443 --name lowkey_vault nagyesta/lowkey-vault:1.8.14 --env LOWKEY_IMPORT_LOCATION=/services/notify/vault.secrets.json.hbs`
> 
> `/services/notify/vault.secrets.json.hbs` is based on the relative file path from my current working directory

Sorry, didn't see this when I replied. Please see this documentation about using volumes with Docker: https://docs.docker.com/storage/volumes/

**Reply by nagyesta** [reply]
Also, you will need to use the `LOWKEY_ARGS` env variable instead of `LOWKEY_IMPORT_LOCATION` and keep `--LOWKEY_IMPORT_LOCATION=/import/vault.secrets.json.hbs` in the value of `LOWKEY_ARGS` as seen here:
https://github.com/nagyesta/spring-boot-local-key-vault-demo/blob/main/docker-compose.yml#L12

This is because of how the entrypoint works here: https://github.com/nagyesta/lowkey-vault/blob/main/lowkey-vault-docker/src/docker/Dockerfile#L7

**Reply by nagyesta** [reply]
I have created a Docker example as well here: https://github.com/nagyesta/lowkey-vault-example-docker
It has basically all current features including the import file with templates.
